languages/python/sqlalchemy-oso/sqlalchemy_oso/oso.py

```diff
@@ -0,0 +1,27 @@
+from oso import Oso, OsoError
+from .auth import register_models
+from .roles2 import OsoRoles
+
+class SQLAlchemyOso(Oso):
+    def __init__(self, sqlalchemy_base):
+        super().__init__()
+
+        # Register all sqlalchemy models on sqlalchemy_base
+        # TODO (dhatch): Not sure this is legit b/c it uses an internal interface?
+        for name, model in sqlalchemy_base._decl_class_registry.items():
+            if name == "_sa_module_registry":
+                continue
+
+            self.register_class(model)
+
+        self.base = sqlalchemy_base
+        self.roles_enabled = False
+
+    def enable_roles(self, user_model, session_maker):
+        self.roles = OsoRoles(oso=self, sqlalchemy_base=self.base, user_model=user_model, session_maker=session_maker)
+        self.roles_enabled = True
+
+    def __getattr__(self, attr):
```

Comment:
Instead of this, use [`@property`](https://docs.python.org/3.6/library/functions.html#property) and define a `roles` property.